### PR TITLE
[CBRD-21734] or_overflow jump causes memory leaks

### PR DESCRIPTION
**Pull request by paulgheorghecristian**
http://jira.cubrid.org/browse/CBRD-21734

Because of the way mr_writeval_string treats errors, I was left with memory leaks. I simply treat the error in the same function to avoid memory leaks.

### CODE REVIEW

File: src/object/object_primitive.c [hunk 1]
```diff
@@ -17456,12 +17459,35 @@ mr_data_writeval_json (OR_BUF * buf, DB_VALUE * value)
...
 
+  if (buf->error_abort)
+    {
+      int estimated_length = mr_data_lengthval_json (value, true);
+
+      /* in case of string compression, buffer will hold
+       * 0xFF int int as first bytes, so we need to take them
+       * in consideration
+       */
+      estimated_length += sizeof (char) + sizeof (int) * 2;
```
Comment (ribram) [inline src/object/object_primitive.c]: Doesn't mr_data_lengthval_json  already count the length header?

Comment (paulgheorghecristian) [reply src/object/object_primitive.c]: you're right

File: src/object/object_primitive.c [hunk 2]
```diff
@@ -17456,12 +17459,29 @@ mr_data_writeval_json (OR_BUF * buf, DB_VALUE * value)
...
     {
       assert (false);
       return ER_FAILED;
     }
 
+  if (buf->error_abort)
+    {
+      int estimated_length = mr_data_lengthval_json (value, true);
+
+      if (estimated_length > buf->endptr - buf->ptr)
```
Comment (eseokoh) [inline src/object/object_primitive.c]: it is better to cast both LHS and RHS as `ptrdiff_t`.
please refer to `mr_data_writeval_set` for an example.

Comment (paulgheorghecristian) [reply src/object/object_primitive.c]: done :)

**Pull request merged**